src/test/kotlin/no/skatteetaten/aurora/gobo/graphql/loadtest/GoboLoadtestSimulationMutationTest.kt

```diff
@@ -0,0 +1,127 @@
+        assertThat(deleteToxiProxyToxicsInput.toxiProxyName).isEqualTo("mokeyToxic")
+        assertThat(deleteToxiProxyToxicsInput.toxicName).isEqualTo("latency_toxic")
+    }
+    @Test
+    fun `verify mutation contents delete timeout toxic`() {
+        val deleteToxiProxyToxicsInput = "src/gatling/resources/mokey_delete_timeout_toxic_mutation.json".jsonCheckFileContentDeleteToxic
+        assertThat(deleteToxiProxyToxicsInput.affiliation).isEqualTo("aup")
+        assertThat(deleteToxiProxyToxicsInput.environment).isEqualTo("utv01")
+        assertThat(deleteToxiProxyToxicsInput.toxiProxyName).isEqualTo("mokeyToxic")
+        assertThat(deleteToxiProxyToxicsInput.toxicName).isEqualTo("timeout_toxic")
+    }
+
+    @Test
+    fun `add delay toxic on toxi-proxy`() {
+        val queryValue = "src/gatling/resources/mokey_add_delay_toxic_mutation.json".jsonInputQuery
+        val inputVariables = "src/gatling/resources/mokey_add_delay_toxic_mutation.json".jsonInputVariables
+        webTestClient.queryGraphQL(queryValue, inputVariables, "test-token").expectStatus().isOk
+            .expectBody()
+            .graphqlDataWithPrefix("addToxiProxyToxic") {
+                graphqlData("toxiProxyName").isEqualTo("mokeyToxic")
+                graphqlData("toxicName").isEqualTo("delay_toxic")
+            }
+            .graphqlDoesNotContainErrors()
+    }
+
+    @Test
+    fun `add latency toxic on toxi-proxy`() {
+        val queryValue = "src/gatling/resources/mokey_add_latency_toxic_mutation.json".jsonInputQuery
+        val inputVariables = "src/gatling/resources/mokey_add_latency_toxic_mutation.json".jsonInputVariables
+        webTestClient.queryGraphQL(queryValue, inputVariables, "test-token").expectStatus().isOk
+            .expectBody()
+            .graphqlDataWithPrefix("addToxiProxyToxic") {
+                graphqlData("toxiProxyName").isEqualTo("mokeyToxic")
+                graphqlData("toxicName").isEqualTo("latency_toxic")
+            }
+            .graphqlDoesNotContainErrors()
+    }
+    @Test
+    fun `add timeout toxic on toxi-proxy`() {
+        val queryValue = "src/gatling/resources/mokey_add_timeout_toxic_mutation.json".jsonInputQuery
+        val inputVariables = "src/gatling/resources/mokey_add_timeout_toxic_mutation.json".jsonInputVariables
+        webTestClient.queryGraphQL(queryValue, inputVariables, "test-token").expectStatus().isOk
+            .expectBody()
+            .graphqlDataWithPrefix("addToxiProxyToxic") {
+                graphqlData("toxiProxyName").isEqualTo("mokeyToxic")
+                graphqlData("toxicName").isEqualTo("timeout_toxic")
+            }
+            .graphqlDoesNotContainErrors()
+    }
+
+    private val String.jsonInputQuery get() = jacksonObjectMapper().readTree(File(this)).at("/query").asText()
+
+    private val String.jsonInputVariables get() = jacksonObjectMapper().readTree(File(this)).at("/variables/input")
+    private val String.jsonCheckFileContentAddToxic get() =
```

Comment:
Hva tenker du om vi gjør sånn:
 ```
private inline fun <reified T> String.toJsonType(): T =
        jacksonObjectMapper().convertValue(jacksonObjectMapper().readTree(File(this)).at("/variables/input"))
```

som kan brukes fra testene på denne måten:
```
val addToxiProxyToxicsInput = "src/gatling/resources/mokey_add_delay_toxic_mutation.json".toJsonType<AddOrUpdateToxiProxyInput>()
```

Da slipper vi egne metoder for hver type.

Reply:
Kult - endret til inline reified.